core/dbt/include/global_project/macros/materializations/incremental/helpers.sql

```diff
@@ -0,0 +1,20 @@
+
+{% macro incremental_upsert(tmp_relation, target_relation, unique_key=none, statement_name="main") %}
+    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) -%}
+    {%- set dest_cols_csv = dest_columns | map(attribute='quoted') | join(', ') -%}
+
+    {%- if unique_key is not none -%}
+    delete
+    from {{ target_relation }}
+    where ({{ unique_key }}) in (
+        select ({{ unique_key }})
+        from {{ tmp_relation.include(schema=False, database=False) }}
```

Comment:
Is this `include()` necessary/even correct for all databases? I think whatever made your `tmp_relation` should be giving you the correct include policy.

Reply:
this is such a great catch! Yes - this macro should definitely expect the `tmp_relation` to already have a valid include policy for the given database.